### PR TITLE
fix: build correct alt dna sequence for deletions

### PR DESCRIPTION
**Pull request by katie-perry**
#447

### CODE REVIEW

**Review by jarbesfeld — DISMISSED** [review]
@katie-perry Nice work! This looks good to me

File: src/variation/gnomad_vcf_to_protein_variation.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def _trim_prefix_or_suffix(
         if aa_match:
             aa_start_pos += aa_match
             aa_alt = aa_alt[aa_match:] if trim_prefix else aa_alt[:-aa_match]
-            aa_ref = aa_ref[aa_match:] if trim_prefix else aa_ref[:-aa_ref]
+            aa_ref = aa_ref[aa_match:] if trim_prefix else aa_ref[:-aa_match]
```
Comment (katie-perry) [inline src/variation/gnomad_vcf_to_protein_variation.py:81]: @korikuzma just wanted your eyes to double check on this - is this correct? It seemed like a bug so I changed it but it wasn't impacting any tests... maybe I should write some tests for this if there aren't already some

**Comment by jarbesfeld** [comment]
@katie-perry Could you add two additional checks in the `test_deletion` test for reading frame 1 negative strand and reading frame 2 positive strand? I also relocated one test that should've been in `test_substitution`. I used chat to help me generate that example, as well as the corresponding submission in ClinVar so I could validate what the protein change was.

**Comment by katie-perry** [comment]
> @katie-perry Could you add two additional checks in the `test_deletion` test for reading frame 1 negative strand and reading frame 2 positive strand? I also relocated one test that should've been in `test_substitution`. I used chat to help me generate that example, as well as the corresponding submission in ClinVar so I could validate what the protein change was.

great idea - I will work on this!

**Review by korikuzma — COMMENTED** [review]
Stated in sprint planning, but posting here too. 

Waiting on @jarbesfeld 's comment for tests to be addressed. Once he re-approves these, please re-request my review!

**Review by jarbesfeld — DISMISSED** [review]
Nice work!

File: src/variation/gnomad_vcf_to_protein_variation.py [hunk 2]
```diff
@@ -299,37 +299,49 @@ def _get_genomic_alt(
...
         :param g_input_alt: Original alteration provided by VCF-like query
+        :param len_g_ref: Length of genomic reference sequence in the input VCF-like
+            expression (edit span length, NOT the codon-aligned window length)
         :param g_end_pos: Genomic end position for codon
         :param alt_type: The type of alteration
         :param genomic_start_ix: The start index for the original genomic start position
+            within ``window_ref``
         :param strand: Strand
-        :param ref: The genomic reference sequence
+        :param window_ref: The codon-aligned genomic reference sequence fetched from
```
Comment (korikuzma) [inline src/variation/gnomad_vcf_to_protein_variation.py]: Consider using more descriptive name 
```suggestion
        :param codon_aligned_ref_seq: The codon-aligned genomic reference sequence fetched from
```

File: tests/test_gnomad_vcf_to_protein.py
Comment (korikuzma) [inline tests/test_gnomad_vcf_to_protein.py:1]: Just want to note I love all the added edge cases.

**Review by korikuzma — COMMENTED** [review]
Didn't review logic (need to whiteboard it when my brain is not fried)

File: src/variation/gnomad_vcf_to_protein_variation.py [hunk 3]
```diff
@@ -579,7 +598,8 @@ async def gnomad_vcf_to_protein(
 
         # Get protein end position
         if alt_type == AltType.DELETION:
-            aa_end_pos = aa_start_pos + (len(aa_ref) - 1)
+            # Inter-residue coordinates use exclusive end
+            aa_end_pos = aa_start_pos + len(aa_ref)
         else:
             aa_end_pos = p_data.pos[1]
```
Comment (korikuzma) [inline src/variation/gnomad_vcf_to_protein_variation.py]: Was wondering if we could simplify this (tests pass with this change, but haven't done a thorough check)

```suggestion
        # Get protein end position
        aa_end_pos = p_data.pos[1]
```

File: tests/test_gnomad_vcf_to_protein.py [hunk 4]
```diff
@@ -256,9 +256,60 @@ def delins_neg():
     return models.Allele(**params)
 
 
+@pytest.fixture(scope="module")
+def brca1_del():
+    """Test fixture for a deletion on a negative strand in the first (0) reading frame (CA003783)"""
```
Comment (korikuzma) [inline tests/test_gnomad_vcf_to_protein.py]: This is a frameshift protein variant, which we currently do not support in VRS, so I don't think we should be returning protein changes for fs at the moment. This is a good edge case for fs though!

File: tests/conftest.py [hunk 5]
```diff
@@ -198,6 +198,71 @@ def protein_insertion():
     return models.Allele(**params)
 
 
+@pytest.fixture(scope="session")
+def pik3r1_deletion():
```
Comment (korikuzma) [inline tests/conftest.py]: This is only used in the gnomad vcf tests, so I think we can move this there

**Review by korikuzma — CHANGES_REQUESTED** [review]
Still looking over logic

File: src/variation/gnomad_vcf_to_protein_variation.py (hunk 3 above)
Comment (korikuzma) [reply src/variation/gnomad_vcf_to_protein_variation.py]: actually, we can remove this and update 

```python
# Trim AA prefixes / suffixes and update the protein start position accordingly
aa_start_pos = p_data.pos[0]
```

to

```python
# Trim AA prefixes / suffixes and update the protein start position accordingly
aa_start_pos, aa_end_pos = p_data.pos
```

File: tests/conftest.py (hunk 5 above)
Comment (korikuzma) [reply tests/conftest.py]: Same with the others

File: tests/test_gnomad_vcf_to_protein.py
Comment (korikuzma) [inline tests/test_gnomad_vcf_to_protein.py:1]: Note for myself: Confirmed protein representations for all (excluding fs) are good

File: tests/test_gnomad_vcf_to_protein.py (hunk 4 above)
Comment (katie-perry) [reply tests/test_gnomad_vcf_to_protein.py]: should I remove it from this test and try to find a different variant for now?

Comment (korikuzma) [reply tests/test_gnomad_vcf_to_protein.py]: No, I think this is good to test and is another bug. So we'd keep the test (remove the fixture) and ensure that frameshifts don't return a VRS object (can include a warning to say frameshifts aren't supported in VRS)

Maybe separate out into new issue?

**Review by korikuzma — CHANGES_REQUESTED** [review]
https://github.com/cancervariants/variation-normalization/pull/647#discussion_r2847002011

If we could remove this before merging, that would be great! I think logic makes sense. I did some clean up work in #649 as I was reviewing. Great work @katie-perry @jarbesfeld

File: tests/test_gnomad_vcf_to_protein.py (hunk 4 above)
Comment (katie-perry) [reply tests/test_gnomad_vcf_to_protein.py]: Created followup issue for this https://github.com/cancervariants/variation-normalization/issues/652

**Review by korikuzma — APPROVED** [review]
